app/src/ui/lib/branch-name-warnings.tsx

```diff
@@ -69,3 +70,15 @@ export function renderBranchNameExistsOnRemoteWarning(
     </Row>
   )
 }
+
+export function renderStashWillBeLostWarning(stash: IStashEntry | null) {
+  if (stash === null) {
+    return null
+  }
+  return (
+    <Row className="warning-helper-text">
+      <Octicon symbol={OcticonSymbol.alert} />
+      <p>Stashed changes on this branch will be lost if it is renamed.</p>
```

Comment:
In reading through this, I was feeling a desire for more specificity. "Lost" vs. "no longer visible in Desktop" seems like a potentially important distinction in deciding how to proceed.
```suggestion
      <p>Your current stashed changes on this branch will no longer be visible in GitHub Desktop if the branch is renamed.</p>
```